src/org/labkey/test/tests/list/CrossfolderListTest.java

```diff
@@ -0,0 +1,208 @@
+        String listName = "subfolder_list_for_metadata";
+        ListDefinition listDef = makeListDef(listName, testFields());
+        listDef.create(createDefaultConnection(), SUBFOLDER_A_PATH);
+
+        // attempt to navigate to the list but in the project level
+        GridPage.beginAt(this, getProjectName(), listName);
+        // expect to land on an error page instead
+        var errPage = new LabkeyErrorPage(getDriver());
+        assertEquals("Expect error explaining that the specified list is not here",
+                "List does not exist in this container", errPage.getErrorHeading());
+    }
+
+    @Test
+    public void testListsWithSameNameInTopAndSubfolder() throws Exception
+    {
+        String listName = "test_name_collision_list";
+
+        // create a simple list in subfolder, insert some values, and grab a column's worth of data from it
+        ListDefinition listDef = makeListDef(listName, testFields());
+        var dGen = listDef.create(createDefaultConnection(), getProjectName());
+        dGen.withGeneratedRows(2);
+        dGen.insertRows();
+        var topStrings = dGen.getRows().stream().map(a-> a.get("stringColumn").toString()).collect(Collectors.toList());
+
+        // create another list with the same name at the project level, insert a little different data and capture the string values
+        ListDefinition listDef2 = makeListDef(listName, testFields());
+        var dgen2 = listDef2.create(createDefaultConnection(), SUBFOLDER_A_PATH);
+        dgen2.withGeneratedRows(3);
+        dgen2.insertRows();
+        var bottomStrings = dgen2.getRows().stream().map(a-> a.get("stringColumn").toString()).collect(Collectors.toList());
+
+        // navigate to the top folder, open the container filter to include the subfolder and ensure only data from this list is shown
+        var topPage = GridPage.beginAt(this, getProjectName(), listName);
+        topPage.getGrid().setContainerFilter(DataRegionTable.ContainerFilterType.CURRENT_AND_SUBFOLDERS);   // include subfolder in filter
+        assertEquals("expect the view in the top folder to only show current list data even with different subfolder list by the same name",
+                new HashSet(topStrings), new HashSet(topPage.getGrid().getColumnDataAsText("String Column")));
+
+        // now view the list from the subfolder and ensure the list contents aren't mixed despite name ambiguity
+        var subfolderPage = GridPage.beginAt(this, SUBFOLDER_A_PATH, listName);
+        subfolderPage.getGrid().setContainerFilter(DataRegionTable.ContainerFilterType.ALL_FOLDERS);
+        assertEquals("expect the view in the top folder to only show current list data even with different subfolder list by the same name",
+                new HashSet(bottomStrings), new HashSet(subfolderPage.getGrid().getColumnDataAsText("String Column")));
+    }
+
+    private ListDefinition makeListDef(String listName, List<FieldDefinition> listColumns)
```

Comment:
nit: We don't tend to use the verb `make` in function/method names. I would recommend renaming to `createListDef` or `getListDef`.